src/savegame_json.cpp

```diff
@@ -3044,6 +3054,34 @@ void item::io( Archive &archive )
         // Some hot/cold items from legacy saves may be inactive
         active = true;
     }
+
+    // Former comestibles that no longer need temperature tracking
+    if( !has_temperature() && ( last_temp_check != calendar::turn_zero || has_own_flag( flag_HOT ) ||
+                                has_own_flag( flag_COLD ) || has_own_flag( flag_FROZEN ) ) ) {
+        bool abort = false;
+        if( active ) {
+            if( temperature_removal_blacklist.count( type->get_id() ) != 0 ) {
```

Comment:
It's called a blacklist but acts as a whitelist. The whitelist is too much extra baggage for a minor migration issue.

Reply:
> It's called a blacklist but acts as a whitelist.

I know; however same is true for `charge_migration_blacklist` and `charge_removal_blacklist` so I'm just following the existing convention.  If you insist, I can rename all of them.

> too much extra baggage

It's insurance against unintended consequences.
Can you provide a strong guarantee that `!has_temperature() && ( last_temp_check != ... )` will never match some other item that we don't actually want to migrate?

Unintended migration stomps over `last_temp_check` and temperature flags. It would lead to incorrect behavior that is very hard to detect by players, because it is difficult to discern from other normal behavior. e.g. for an item that ought to have been `HOT` one might think that it had cooled down naturally. I'd rather try to avoid that.